aea/helpers/async_utils.py

```diff
@@ -477,3 +458,206 @@ def __init__(self, getters: List[Tuple[Callable[[Any], None], Callable]]):
         super().__init__(
             [self._make_getter(handler, getter) for handler, getter in getters]
         )
+
+
+ready_future: Future = Future()
+ready_future.set_result(None)
+
+
+class Runnable(ABC):
+    """Abstract Runnable class."""
+
+    def __init__(self, loop=None, threaded=False) -> None:
+        """
+        Init runnable.
+
+        :param loop: asyncio event loop to use.
+        :param threaded: bool. start in thread if True.
+
+        :return: None
+        """
+        if loop and threaded:
+            raise ValueError(
+                "You can not set a loop in threaded mode, cause own loop will be created"
+            )
+        self._loop = loop
+        self._threaded = threaded
+        self._task: Optional[asyncio.Task] = None
+        self._thread: Optional[Thread] = None
+        self._completed_event: Optional[asyncio.Event] = None
+        self._got_result = False
+        self._was_cancelled = False
+        self._is_running: bool = False
+
+    def start(self) -> bool:
+        """
+        Start runnable.
+
+        :return: bool started or not.
+        """
+        if self._task and not self._task.done():
+            logger.debug("already running")
```

Comment:
Debug statement could be made more explicit.

Reply:
done